onadata/apps/api/viewsets/open_data_viewset.py

```diff
@@ -0,0 +1,287 @@
+        '''
+        Retrieve columns headers that are valid in tableau.
+        '''
+        tableau_colulmn_headers = []
+
+        def append_to_tableau_colulmn_headers(header, question_type=None):
+            quest_type = 'string'
+            if question_type:
+                quest_type = question_type
+
+            # alias can be updated in the future to question labels
+            tableau_colulmn_headers.append({
+                'id': header,
+                'dataType': quest_type,
+                'alias': header
+            })
+
+        # using nested loops to determine what valid data types to set for
+        # tableau.
+        for header in self.xform_headers:
+            for quest_name, quest_type in self.flattened_dict.items():
+                if header == quest_name or header.endswith('_%s' % quest_name):
+                    append_to_tableau_colulmn_headers(header, quest_type)
+                    break
+            else:
+                append_to_tableau_colulmn_headers(header)
+
+        return tableau_colulmn_headers
+
+    def create(self, request, *args, **kwargs):
+        if request.user.is_anonymous():
+            return Response(
+                'Authentication credentials required.',
+                status.HTTP_400_BAD_REQUEST
+            )
+
+        results = self.get_data()
+        if results.error:
+            return Response(results.message, status.HTTP_400_BAD_REQUEST)
+
+        if results.data:
+            serializer = OpenDataSerializer(data=results.data)
+
+            if serializer.is_valid():
+                _open_data = serializer.save()
+                if _open_data:
+                    return Response(
+                        data={
+                            'message': 'Record was successfully created.',
+                            'uuid': _open_data.uuid
+                        },
+                        status=status.HTTP_201_CREATED
+                    )
+            else:
+                return Response(
+                    str(serializer.errors), status.HTTP_400_BAD_REQUEST
+                )
+
+    @detail_route(methods=['GET'])
+    def data(self, request, **kwargs):
+        self.object = self.get_object()
+        # get greater than value and cast it to an int
+        gt = request.query_params.get('gt_id')
+        gt = gt and parse_int(gt)
+
+        data = []
+        if isinstance(self.object.content_object, XForm):
+            if not self.object.active:
+                return Response(status=status.HTTP_404_NOT_FOUND)
+
+            xform = self.object.content_object
+            qs_kwargs = {'xform': xform}
+            if gt:
+                qs_kwargs.update({'id__gt': gt})
+
+            instances = Instance.objects.filter(**qs_kwargs)
+            csv_df_builder = CSVDataFrameBuilder(
+                xform.user.username, xform.id_string, include_images=False
+            )
+            data = csv_df_builder._format_for_dataframe(
```

Comment:
I have added a `MAX_INSTANCES_PER_REQUEST` constant whose value is the maximum number of records returned per request. The current value is 1000.